src/MongoDB.Driver/Linq/Processors/AccumulatorBinder.cs

```diff
@@ -207,6 +207,19 @@ private Expression GetAccumulatorArgument(Expression node)
                 return select.Selector;
             }
 
+            if (allowWhereExpression)
```

Comment:
It appears like this is _always_ only for a count expression. If that's the case, let's go ahead and just make this a different function rather than overload this one. GetCountAccumulatorArgument() perhaps. My concern is that if we get another accumulator that also supports a predicate, then we'll just call this and it likely isn't correct. For instance, First and Last also support predicates, but we cannot support those predicates like this (or at all because MongoDB doesn't have a mechanism for those).

Reply:
Done